runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,391 @@
+                            }
+
+                            txn.commit();
+                            return null;
+                        }
+
+                        log.debug("[{}] {} {}", clientName, "No matching destination found for:", destination);
+                        return null;
+                    }
+                    throw new NoSuchElementException(String.format("Record not found for group: %s", logicalGroup));
+                } catch (TransactionAbortedException tae) {
+                    log.error("[{}] {}: {}", clientName, "Unable to remove destination, will be retried", tae);
+                    throw new RetryNeededException();
+                }
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("[{}] {}: {}", clientName, "Unable to remove destination", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Remove a list of destinations from the LogReplicationModelMetadataTable for a specific group.
+     *
+     * @param logicalGroup The group of tables that are replicated, represented as a string.
+     * @param remoteDestinations The cluster from which the logicalGroup should be removed from all future replication.
+     */
+    public void removeDestination(String logicalGroup, List<String> remoteDestinations) {
+        Preconditions.checkArgument(isValid(logicalGroup), String.format("[%s] %s", clientName, "logicalGroup is null or empty."));
+        Preconditions.checkArgument(isValid(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations is null or empty."));
+        Preconditions.checkArgument(hasNoNullOrEmptyElements(remoteDestinations), String.format("[%s] %s", clientName, "remoteDestinations contains null or empty elements."));
+        Set<String> removeRemoteDestinationsSet = deduplicate(remoteDestinations);
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = buildClientDestinationInfoKey(logicalGroup);
+                    DestinationInfoVal clientDestinations = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+
+                    if (clientDestinations != null) {
+                        Set<String> clientDestinationIdsSet = new HashSet<>(clientDestinations.getDestinationIdsList());
+                        List<String> clientDestinationIdsList = clientDestinationIdsSet.stream()
+                                .filter(i -> !removeRemoteDestinationsSet.contains(i))
+                                .collect(Collectors.toList());
+
+                        if (clientDestinationIdsList.size() == clientDestinationIdsSet.size()) {
+                            log.debug("[{}] {} {}", clientName, "No matching destinations found for:", removeRemoteDestinationsSet);
```

Comment:
this will only log the object reference of the remote destinations set.  We will have to iterate through the set and print each destination.

Reply:
I see, I was using print statements for debugging the format and the implicit .toString() call was handling the conversion, I suppose calling it explicitly here would work as well instead of iterating through the set?